File: examples/strongscript/3.ts
class A {
	constructor() {}
	getN():!number { return 2; }
}

class C extends A {
	constructor() {super()}
	getN():!number { return 4; }
}

class D {
	constructor(public ref : !A, public anyref) {}
	main():!number { 
		this.ref = new C();
		this.anyref = this.ref;
		return this.anyref; }
}

File: examples/reticulated_python/3.py
class A:
	def getN()->Int:
		return 2

class C(A):
	def getN()->Int:
		return 4

@fields({'ref' : A, 'anyref' : Dyn})
class D:
	ref = A()
	anyref = A()
	def main(self:D) -> C:
		self.ref = C()
		self.anyref = self.ref
		return self.anyref


D().main()
